boards/xtensa/esp32/CMakeLists.txt

```
@@ -4,28 +4,44 @@
 if(CONFIG_BOOTLOADER_ESP_IDF)
   include(ExternalProject)
 
+  if(NOT (${CMAKE_HOST_SYSTEM_NAME} STREQUAL Windows))
+    set(espidf_tool   PATH=$ENV{PATH}:${ESPRESSIF_TOOLCHAIN_PATH}/bin)
+  endif()
+
   ## we use hell-world project, but I think any can be used.
-  set(espidf_src_dir   ${ESP_IDF_PATH}/examples/get-started/hello_world/)
+  set(espidf_components_dir   ${ESP_IDF_PATH}/components)
   set(espidf_prefix    ${CMAKE_BINARY_DIR}/esp-idf)
   set(espidf_build_dir ${espidf_prefix}/build)
 
   ExternalProject_Add(
     EspIdfBootloader
     PREFIX ${espidf_prefix}
-    SOURCE_DIR ${espidf_src_dir}
-    BINARY_DIR ${espidf_src_dir}
+    SOURCE_DIR ${espidf_components_dir}/bootloader/subproject
+    BINARY_DIR ${espidf_build_dir}/bootloader
     CONFIGURE_COMMAND ""    # Skip configuring the project, e.g. with autoconf
     BUILD_COMMAND
-    PATH=$ENV{PATH}:${ESPRESSIF_TOOLCHAIN_PATH}/bin
-    make
-    IDF_PATH=${ESP_IDF_PATH}
-    BUILD_DIR_BASE=${espidf_build_dir}
-    SDKCONFIG=${espidf_build_dir}/sdkconfig
-    defconfig bootloader partition_table
+    ${espidf_tool}
+    cmake -GNinja
+    -S ${espidf_components_dir}/bootloader/subproject
+    -B ${espidf_build_dir}/bootloader -DSDKCONFIG=${espidf_build_dir}/sdkconfig
+    -DIDF_PATH=${ESP_IDF_PATH} -DIDF_TARGET=${CONFIG_BOARD}
+    -DPYTHON_DEPS_CHECKED=1
+    && cd ${espidf_build_dir}/bootloader && cmake --build .
```

This should be divided into:
- `CONFIGURE_COMMAND`
- `BUILD_COMMAND`

also use `${CMAKE_COMMAND}` instead of `cmake` and `${CMAKE_GENERATOR}` instead of Ninja, to so that same generator is used for both current project and the external one.

There is no need for the `cd` command, as the build command will execute fot the `BINARY_DIR` var.

```suggestion
    CONFIGURE_COMMAND
    ${espidf_tool}
    ${CMAKE_COMMAND} -G${CMAKE_GENERATOR}
    -S ${espidf_components_dir}/bootloader/subproject
    -B ${espidf_build_dir}/bootloader -DSDKCONFIG=${espidf_build_dir}/sdkconfig
    -DIDF_PATH=${ESP_IDF_PATH} -DIDF_TARGET=${CONFIG_BOARD}
    -DPYTHON_DEPS_CHECKED=1
    BUILD_COMMAND
    ${CMAKE_COMMAND} --build .
```